lib/elixir/lib/module/checker.ex

```diff
@@ -3,53 +3,65 @@ defmodule Module.Checker do
 
   @moduledoc false
 
-  def verify(module_map, binary, cache) do
-    warnings = warnings(module_map, cache)
-    binary = chunk(binary, module_map)
+  def verify(map, chunks, cache) do
+    # TODO: Add ExCk chunk to state so that ParallelChecker does not
+    #       have to load from disk again
+
+    warnings = warnings(map, cache)
+    binary = add_chunk(chunks, map)
     {binary, warnings}
   end
 
-  defp chunk(binary, module_map) do
-    checker_chunk = build_chunk(module_map)
-    {:ok, _module, all_chunks} = :beam_lib.all_chunks(binary)
-    {:ok, binary} = :beam_lib.build_module([checker_chunk | all_chunks])
+  defp add_chunk(chunks, map) do
+    checker_chunk = build_chunk(map)
+    chunks = [checker_chunk | :lists.keydelete('ExCk', 1, chunks)]
+    {:ok, binary} = :beam_lib.build_module(chunks)
     binary
   end
 
-  defp build_chunk(module_map) do
-    exports = ParallelChecker.definitions_to_exports(module_map.definitions)
-    deprecated = :maps.from_list(module_map.deprecated)
+  defp build_chunk(map) do
+    exports = ParallelChecker.definitions_to_exports(map.definitions)
+    deprecated = :maps.from_list(map.deprecated)
 
-    contents =
+    exports =
       Enum.map(exports, fn {function, kind} ->
         reason = :maps.get(function, deprecated, nil)
         {function, {kind, reason}}
       end)
 
-    {'ExCk', :erlang.term_to_binary({:elixir_checker_v1, Enum.sort(contents)})}
+    contents = %{
+      exports: Enum.sort(exports),
+      compile_opts: filter_compile_opts(map.compile_opts)
```

Comment:
Perhaps it is worth storing it already has `no_warn_undefined`, since we always precompute it anyway?